File: Chronos.Core/Native/WindowArgs.cs
namespace Chronos.Core.Native;

public class WindowArgs : EventArgs
{
    public WindowArgs(
        nint hookHandle,
        uint eventType,
        nint windowHandle,
        int objectId,
        int childId,
        uint eventThreadId,
        uint eventTime)
    {
        Handle = hookHandle;
        EventType = eventType;
        WindowHandle = windowHandle;
        ObjectId = objectId;
        ChildId = childId;
        EventThreadId = eventThreadId;
        EventTime = eventTime;
    }

    public nint Handle { get; }

    public uint EventType { get; }

    public nint WindowHandle { get; }

    public int ObjectId { get; }

    public int ChildId { get; }

    public uint EventThreadId { get; }

    public uint EventTime { get; }
}

File: Chronos.Core/Native/WindowArgs.fs
namespace Chronos.Core.Native

open System

type WindowArgs
    (
        handle: nativeint,
        eventType: uint32,
        windowHandle: nativeint,
        objectId: int,
        childId: int,
        eventThreadId: uint32,
        eventTime: uint32
    ) =

    inherit EventArgs()

    member val Handle = handle with get
    member val EventType = eventType with get
    member val WindowHandle = windowHandle with get
    member val ObjectId = objectId with get
    member val ChildId = childId with get
    member val EventThreadId = eventThreadId with get
    member val EventTime = eventTime with get
